test/site/_collections/mocks/collection.js

```diff
@@ -0,0 +1,21 @@
+module.exports = class MockCollection {
+  constructor(items) {
+    this.items = items;
+  }
+
+  getAll() {
+    return this.items;
+  }
+
+  getFilteredByGlob() {
+    return this.items;
+  }
+
+  getFilteredByTag() {
+    return this.items;
```

Comment:
Instead of removing the call to this method from the test's scope (by mocking it here), I'd rather see us extract the relevant part of the logic we want to test to a separate function and test that.